Libraries/PushNotificationIOS/RCTPushNotificationManager.m

```diff
@@ -57,6 +57,24 @@ + (UILocalNotification *)UILocalNotification:(id)json
 
 @implementation RCTPushNotificationManager
 
+static NSDictionary *formatLocalNotification(UILocalNotification *notification)
+{
+  NSMutableDictionary *formattedLocalNotification = [NSMutableDictionary dictionary];
+  if (notification.fireDate) {
+    NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    [formatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"];
+    NSString *fireDateString = [formatter stringFromDate:notification.fireDate];
+    formattedLocalNotification[@"fireDate"] = fireDateString;
+  }
+  formattedLocalNotification[@"alertAction"] = RCTNullIfNil(notification.alertAction);
+  formattedLocalNotification[@"alertBody"] = RCTNullIfNil(notification.alertBody);
+  formattedLocalNotification[@"applicationIconBadgeNumber"] = @(notification.applicationIconBadgeNumber);
+  formattedLocalNotification[@"category"] = RCTNullIfNil(notification.category);
+  formattedLocalNotification[@"soundName"] = RCTNullIfNil(notification.soundName);
+  formattedLocalNotification[@"userInfo"] = RCTNullIfNil(RCTJSONClean(notification.userInfo));
+  return [NSDictionary dictionaryWithDictionary:formattedLocalNotification];
```

Comment:
Every NSMutableDictionary is also a NSDictionary, so you can just return `formattedLocalNotification` here. (Technically, the object is still an NSMutableDictionary, so any caller of this method could cast and mutate it, but this function doesn't need to care about it).